backend/src/main/java/com/example/backend/data_access/route/RouteDAO.java

```diff
@@ -180,4 +186,43 @@ public HashMap<String, HashSet<String>> getStopTagsToRouteTags() {
         return stopTagsToRouteTags;
     }
 
+    @Override
+    public HashMap<String, ArrayList<Location>> getRouteShapes() {
+        HashMap<String, ArrayList<Location>> routeShapes = new HashMap<>();
+
+        // Used BufferedReader for massively improved performance
+        try (BufferedReader reader = new BufferedReader(new FileReader(shapesCsvFilename))) {
+            String line;
+            // Skip first row (headers)
+            reader.readLine();
+
+            while ((line = reader.readLine()) != null) {
+                String[] row = line.split(",");
+                // Get the routeId, latitude and longitude from the row
+                String routeId = row[1];
```

Comment:
I'm confused. In the csv file, row[1] is marked as shape id. However, the variable is called route id. Which is it?

Reply:
It's supposed to be route id, the shape id column on the csv is a remnant from the original dataset which I forgot to change